### PR TITLE
Lightning

### PR DESCRIPTION
**Pull request by nymkappa**
### Testing

* Make sure you `state.schema_version` is set to 22 (current master)
* Make sure to get rid of all LN related tables in your db (`channels`, `lightning_stats`, `nodes`, `node_stats`)
* Make sure to delete `state.last_node_stats` entry
* Run the backend and pray

### CODE REVIEW

**Comment by softsimon** [comment]
Ready to test :)

**Comment by wiz** [comment]
please rebase and increment db schema version

**Comment by knorrium** [comment]
@softsimon can you update the lock file?

**Comment by wiz** [comment]
can you please fix page title and SEO metadata for /lightning

File: backend/src/api/explorer/channels.routes.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      res.json(channel);
+    } catch (e) {
+      res.status(500).send(e instanceof Error ? e.message : e);
+    }
+  }
+
+  private async $getChannelsForNode(req: Request, res: Response) {
+    try {
+      if (typeof req.query.public_key !== 'string') {
+        res.status(501).send('Missing parameter: public_key');
```
Comment (wiz) [inline backend/src/api/explorer/channels.routes.ts:42]:
```suggestion
        res.status(400).send('Missing parameter: public_key');
```

File: backend/src/api/explorer/channels.routes.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      res.json(channels);
+    } catch (e) {
+      res.status(500).send(e instanceof Error ? e.message : e);
+    }
+  }
+
+  private async $getChannelsByTransactionIds(req: Request, res: Response) {
+    try {
+      if (!Array.isArray(req.query.txId)) {
+        res.status(500).send('Not an array');
```
Comment (wiz) [inline backend/src/api/explorer/channels.routes.ts:60]:
```suggestion
        res.status(400).send('Not an array');
```

File: backend/src/api/explorer/general.routes.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    app
+      .get(config.MEMPOOL.API_URL_PREFIX + 'lightning/search', this.$searchNodesAndChannels)
+      .get(config.MEMPOOL.API_URL_PREFIX + 'lightning/statistics/latest', this.$getGeneralStats)
+      .get(config.MEMPOOL.API_URL_PREFIX + 'lightning/statistics', this.$getStatistics)
+    ;
+  }
+
+  private async $searchNodesAndChannels(req: Request, res: Response) {
+    if (typeof req.query.searchText !== 'string') {
+      res.status(501).send('Missing parameter: searchText');
```
Comment (wiz) [inline backend/src/api/explorer/general.routes.ts:19]:
```suggestion
      res.status(400).send('Missing parameter: searchText');
```

**Review by wiz — CHANGES_REQUESTED** [review]
Tested ACK @ v2.4.1-dev [[18030ba3](https://github.com/mempool/mempool/commit/18030ba3)]

A few very minor requested changes, but tested and reviewed OK :rocket:

**Review by wiz — APPROVED** [review]
Tested ACK @ v2.4.1-dev [[18030ba3](https://github.com/mempool/mempool/commit/18030ba3)]

**Comment by wiz** [comment]
CI found a bug in the liquid block page, should fix later in separate PR 

Screen Shot 2022-07-10 at 15 07 18

**Pull request merged**